pootle/apps/pootle_app/management/commands/create_project.py

```diff
@@ -0,0 +1,171 @@
+            help="Project name",
+        )
+        parser.add_argument(
+            "--filetype",
+            action="append",
+            dest="filetypes",
+            choices=format_names,
+            default=[],
+            help="File types: {0}. Default: po".format(
+                 " | ".join(format_names)),
+        )
+        mapping_group = parser.add_mutually_exclusive_group(required=True)
+        mapping_group.add_argument(
+            "--preset-mapping",
+            action="store",
+            dest="preset_mapping",
+            choices=FS_PRESETS.keys(),
+            help="Preset filesystem layout: {0}".format(
+                 " | ".join(FS_PRESETS.keys())),
+        )
+        mapping_group.add_argument(
+            "--mapping",
+            action="store",
+            dest="mapping",
+            help="Custom filesystem layout",
+        )
+        parser.add_argument(
+            "--fs-type",
+            action="store",
+            dest="fs_type",
+            default="localfs",
+            choices=fs_plugins_names,
+            help="Filesystem type: {0}".format(" | ".join(fs_plugins_names)),
+        )
+        parser.add_argument(
+            "--fs-url",
+            action="store",
+            dest="fs_url",
+            default="",
+            help="Filesystem path or URL.",
+        )
+        parser.add_argument(
+            "--source-language",
+            action="store",
+            dest="sourcelang",
+            default="en",
+            help=("Source language. Examples: [en | es | fr | ...]."
+                  "Default: %(default)s"),
+        )
+        parser.add_argument(
+            "--report-email",
+            action="store",
+            default="",
+            dest="contact",
+            help="Contact email for reports. Example: admin@mail.com.",
+        )
+        parser.add_argument(
+            "--disabled",
+            action="store_true",
+            dest="disabled",
+            help="Does the project start disabled?",
+        )
+        parser.add_argument(
+            "--checkstyle",
+            action="store",
+            dest="checkstyle",
+            choices=PROJECT_CHECKERS.keys(),
+            default="standard",
+            help="Quality check styles. Example: {0}. Default: %(default)s"
+                 .format(" | ".join(PROJECT_CHECKERS.keys())),
+        )
+
+    @transaction.atomic
+    def handle(self, **options):
+        """Immitates the behaviour of the admin interface
+        for creating a project from command line.
+        """
+        self.check_project_options(**options)
+        self.set_project_config(
+            self.create_project(**options), **options)
+
+    def check_project_options(self, **options):
+        if options["contact"]:
+            try:
+                validate_email(options["contact"])
+            except ValidationError as e:
+                if options["traceback"]:
+                    traceback.print_exc()
```

Comment:
I am wondering if the traceback on validation might be needed in any scenario. I mean, it either fails the validation or passes it, seeing the traceback does not convey any useful information, afaict, besides the one that the commanderror with the validation error message will already provide.

Reply:
i think the problem is that when you catch and reraise with CommandError - it swallows the original error and makes the `--traceback` option useless - i think its better to have the option to print the traceback should you require

Reply:
Ok.